app/scripts/metamask-controller.js

```diff
@@ -5567,11 +5574,26 @@ export default class MetamaskController extends EventEmitter {
               },
             },
           ),
-        requestPermissionsForOrigin:
+        grantPermittedChainsPermissionIncremental: (chainIds) =>
+          this.permissionController.grantPermissionsIncremental({
+            subject: { origin },
+            approvedPermissions: {
+              [PermissionNames.permittedChains]: {
+                caveats: [
+                  CaveatFactories[CaveatTypes.restrictNetworkSwitching](
+                    chainIds,
+                  ),
+                ],
+              },
+            },
+          }),
+        requestPermissionsForOrigin: (requestedPermissions) => {
           this.permissionController.requestPermissions.bind(
             this.permissionController,
             { origin },
```

Comment:
this shouldn't be a `.bind` anymore and also needs to return the value of the call

```suggestion
        requestPermissionsForOrigin: (requestedPermissions) => {
          return this.permissionController.requestPermissions(
            { origin },
```